### DISCUSSION TITLE
Is the output supposed to cut off and then list related excerpts from documents, or is something bugged?

### DISCUSSION THREAD

**Discussion by randyb100** [imartinez/privateGPT#870] [Q&A]
When I ask it a question, it will give me a response and then cut off mid-sentence, then list excerpts from documents that are somewhat relevant. I asked it to write a python script and this is what it gave me:

`> Answer (took 418.61 s.):
 Here��s a sample Python script you could run on your Raspberry Pi or any other machine, which sends out UDP packets containing information about viewers' IP addresses. You'll need two things for this project—an Internet Service Provider (ISP) that provides the necessary bandwidth to send and receive data over an internet connection; 
and OpenSSL library with support of TLSv1_2 encryption protocol as it is required by PiVPN-ng package used in RaspberryPiOS: you can download from https://www.openssl.org/

> source_documents/525876160-Press-black-hat-Python-2nd-edition-Early-access-1718501129.pdf:
Owning the Network with Scapy    65
4 def extract_content(Response, content_name='image'):
    pass
class Recapper:
    def __init__(self, fname):
        pass
 5 def get_responses(self):
        pass
 6 def write(self, content_name):
        pass
if __name__ == '__main__':
    pfile = os.path.join(PCAPS, 'pcap.pcap')
    recapper = Recapper(pfile)
    recapper.get_responses()
    recapper.write('image')
This is the main skeleton logic of the entire script, and we’ll add in the

> source_documents/525876160-Press-black-hat-Python-2nd-edition-Early-access-1718501129.pdf:
sender.sendto(bytes(MESSAGE, 'utf8'), (str(ip), 65212))
class Scanner: 3
    def __init__(self, host):
        self.host = host
        if os.name == 'nt':
            socket_protocol = socket.IPPROTO_IP
        else:
            socket_protocol = socket.IPPROTO_ICMP
        self.socket = socket.socket(socket.AF_INET,
                                        socket.SOCK_RAW, socket_protocol)
        self.socket.bind((host, 0))

> source_documents/525876160-Press-black-hat-Python-2nd-edition-Early-access-1718501129.pdf:
print("[remotehost] [remoteport] [receive_first]")
        print("Example: ./proxy.py 127.0.0.1 9000 10.12.132.1 9000 True")
Black Hat Python (Early Access) © 2021 by Justin Seitz and Tim Arnold

> source_documents/525876160-Press-black-hat-Python-2nd-edition-Early-access-1718501129.pdf:
6   Chapter 1
you decide to share it. The Python community has a such a guideline, called 
PEP 8. You can read the full PEP 8 guide here: https://www.python.org/dev/peps/
pep-0008/.
The examples in this book generally follow PEP 8, with a few differ-
ences. You’ll see that the code in this book follows a pattern like this:
1 from lxml import etree
from subprocess import Popen
2 import argparse
import os
3 def get_ip(machine_name):  
    pass  
4 class Scanner:
    def __init__(self):
`

It starts writing what it's doing then cuts off. I don't know if it's even related. Other questions are even worse, and it'll just list snippets of documents that are somewhat related to what I'm asking it. It's all over the place.

**Comment by ravictlp** [comment]
I also get this problem too, sometimes it doesn't complete the answer other times, it is able to answer fine. I don't really know why this happens. I think this is a problem of the gpt4All model, because I tried using Vicuna 13b one time with privateGPT (my laptop doesn't have a nvidia gpu so it was very slow using vicuna) and it usually gave me coherent answers.
